src/models/syncRule.ts

```diff
@@ -133,11 +149,146 @@ export default class SyncRule {
     }
 
     public updateMetadataIds(metadataIds: string[]): SyncRule {
+        if (_.isEqual(this.metadataIds, metadataIds)) {
+            return SyncRule.build({
+                ...this.syncRule,
+                builder: {
+                    ...this.syncRule.builder,
+                    metadataIds,
+                },
+            });
+        } else {
+            // When metadataIds really has changed we should reset
+            // useDefaultIncludeExclude and metadataIncludeExcludeRules
+            return SyncRule.build({
+                ...this.syncRule,
+                builder: {
+                    ...this.syncRule.builder,
+                    metadataIds,
+                    useDefaultIncludeExclude: true,
+                    metadataIncludeExcludeRules: undefined,
+                },
+            });
+        }
+    }
+
+    public markToUseDefaultIncludeExclude(): SyncRule {
+        return SyncRule.build({
+            ...this.syncRule,
+            builder: {
+                ...this.syncRule.builder,
+                useDefaultIncludeExclude: true,
+                metadataIncludeExcludeRules: undefined,
+            },
+        });
+    }
+
+    public markToNotUseDefaultIncludeExclude(models: Array<typeof D2Model>): SyncRule {
+        const metadataIncludeExcludeRules: MetadataIncludeExcludeRules = models.reduce(
+            (accumulator: any, model: typeof D2Model) => ({
+                ...accumulator,
+                [model.getMetadataType()]: {
+                    includeRules: model.getIncludeRules().map(array => array.join(".")),
+                    excludeRules: model.getExcludeRules().map(array => array.join(".")),
+                },
+            }),
+            {}
+        );
+
+        return SyncRule.build({
+            ...this.syncRule,
+            builder: {
+                ...this.syncRule.builder,
+                useDefaultIncludeExclude: false,
+                metadataIncludeExcludeRules,
+            },
+        });
+    }
```

Comment:
These two methods could be joined and/or simplified? We have two methods to touch the same two properties.

I'm also not sure if I would make the toggle independent of the customized rules, so the user does not loose them when enabling or disabling the behavior.

Reply:
Two methods are necessary because the behaviour and arguments are different in every case. 
Customization rules are dependent on toggle because if the toggle is marked to true, the user can not customize and should lose previous customizations. And when the toggle is marked to false, It's necessary to set metadataIncludeExcludeRules as hardcoded in models.
This behaviour allows to synchronization process using hardcoded exclude include rules in models when metadataIncludeExcludeRules are undefined (for example on-demand sync)